smithy-model/src/main/java/software/amazon/smithy/model/traits/RetryableTrait.java

```diff
@@ -52,12 +52,16 @@ public boolean getThrottling() {
 
     @Override
     public Builder toBuilder() {
-        return builder().throttling(throttling);
+        return builder().sourceLocation(getSourceLocation()).throttling(throttling);
     }
 
     @Override
     protected Node createNode() {
-        return throttling ? Node.objectNode().withMember(THROTTLING, true) : Node.objectNode();
+        ObjectNode.Builder nodeBuilder = Node.objectNodeBuilder().sourceLocation(getSourceLocation());
```

Comment:
I'm unsure about the need for this propagation. 
```
/**
 * The result of toNode is used for hashCodes and equality. Subclasses
 * must implement createNode to turn the trait into a Node. This is then
 * cached for subsequent retrievals.
 *
 * @return Returns the trait as a node.
 */
protected abstract Node createNode();
```
And Node's equals doesn't look at sourceLocation. So does propagating sourceLocation matter?

Reply:
Yes we need it set here because we use SourceLocation in places like the NodeValidationVisitor (e.g., https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/java/software/amazon/smithy/model/validation/node/PatternTraitPlugin.java#L43)